.github/workflows/ci-cd.yml

```diff
@@ -173,18 +168,8 @@ jobs:
       with:
         allow-prereleases: true
         python-version: ${{ matrix.pyver }}
-    - name: Get pip cache dir
-      id: pip-cache
-      run: |
-        echo "dir=$(pip cache dir)" >> "${GITHUB_OUTPUT}"
-      shell: bash
-    - name: Cache PyPI
-      uses: actions/cache@v4.1.2
-      with:
-        key: pip-ci-${{ runner.os }}-${{ matrix.pyver }}-${{ matrix.no-extensions }}-${{ hashFiles('requirements/*.txt') }}
-        path: ${{ steps.pip-cache.outputs.dir }}
-        restore-keys: |
-            pip-ci-${{ runner.os }}-${{ matrix.pyver }}-${{ matrix.no-extensions }}-
```

Comment:
I'm worried it won't take `matrix.no-extensions` into consideration

Reply:
When it runs I think we can look at the cache key it generates and see if it has no-ext